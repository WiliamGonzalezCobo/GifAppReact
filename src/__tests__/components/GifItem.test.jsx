const { render, screen } = require("@testing-library/react");
const { scryRenderedComponentsWithType } = require("react-dom/test-utils");
const { GifItem } = require("../../components/GifItem");

describe('Pruebas en <GifItem/>', () => {

    const title = 'mi titulo';
    const url = 'http://miurl.com/';

    test('should be match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}></GifItem>)
        expect(container).toMatchSnapshot();
    })

    test('should be show url and alt', () => {
        render(<GifItem title={title} url={url}></GifItem>)

        //expect(screen.getByRole('img').src).toBe(url)
        //expect(screen.getByRole('img').alt).toBe(title)

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })

    test('should be exist text title', () => {
        render(<GifItem title={title} url={url}></GifItem>)
        expect(screen.getByText(title)).toBeTruthy();
    })
})