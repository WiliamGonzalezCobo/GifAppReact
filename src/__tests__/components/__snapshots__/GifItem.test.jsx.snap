// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="mi titulo"
      src="http://miurl.com/"
    />
    <p>
      mi titulo
    </p>
  </div>
</div>
`;
